awkward/array/table.py

```diff
@@ -408,7 +404,7 @@ def _length(self):
 
     def _index(self):
         if self._view is None:
-            return None
+            return slice(self._length())
```

Comment:
This is unrelated to the new feature, but it's definitely an improvement. Thanks for DRYing this WET code.